### PR TITLE
Improved trouble shooting guide

### PR DESCRIPTION
**Pull request by cheld**
A couple of issues are currently around authentication. I got the impression that user got confused where to start the kubectl proxy, which url to type into browser and they could not distinguish if they authentication problems with apiserver or service account.
Created a diagram, hopefully explains a bit better.

### CODE REVIEW

**Comment by cheld** [comment]
@digitalfishpond can you review?

File: docs/user-guide/troubleshooting.md [hunk 1]
```diff
@@ -8,16 +8,54 @@ Do not move it without providing redirects.
...
 
-    Get https://1.2.3.4/api/v1/replicationcontrollers: x509: failed to load system roots and no roots provided
+## Authentication to the Kubernetes API Server
 
-This means that Dashboard failed to authenticate to the API server. Before
+If your Kubernetes cluster is not configured correctly, it may fail to contact
+the API server. Before
 explaining the solution, it is useful to review how the dashboard discovers
 and authenticates with the API server.
+A number of components are involved in the authentication process and the first step is narrow
```
Comment (digitalfishpond) [inline docs/user-guide/troubleshooting.md]: ...step is **to** narrow...

File: docs/user-guide/troubleshooting.md [hunk 2]
```diff
@@ -8,16 +8,54 @@ Do not move it without providing redirects.
...
+down the source of the problem, mainly if you have problems with user authentication
```
Comment (digitalfishpond) [inline docs/user-guide/troubleshooting.md]: ...problem, **namely whether it is a problem with user authentication or with service authentication**.

File: docs/user-guide/troubleshooting.md [hunk 3]
```diff
@@ -8,16 +8,54 @@ Do not move it without providing redirects.
...
+|  +------------+  |      with username &          |                                                    |
+|  browser         |      password only            |                                                    |
+|                  |                               |                                                    |
+|                  |                               |                                                    |
++------------------+                               +----------------------------------------------------+
+
+```
+__User authentication__
+
+Browsers can only authenticate with username and password. If youre apiserver is setup accordingly,
```
Comment (digitalfishpond) [inline docs/user-guide/troubleshooting.md]: ...If **your** apiserver is **set up** **correctly**...

File: docs/user-guide/troubleshooting.md [hunk 4]
```diff
@@ -8,16 +8,54 @@ Do not move it without providing redirects.
...
+ then you can access directly with `https://<master>/ui`. Otherwise, you can start a local proxy
```
Comment (digitalfishpond) [inline docs/user-guide/troubleshooting.md]: ...access it directly via...

File: docs/user-guide/troubleshooting.md [hunk 5]
```diff
@@ -8,16 +8,54 @@ Do not move it without providing redirects.
...
+ with `kubectl proxy` and access Dashboard with the URL `http://localhost:8001`.
+
+__Service authentication__
+
+If you have successfully authenticated, then apiserver proxy will forward the request to dashboard backend. However, the dashboard backend will need information from apiserver and has to authenticate with apiserver again.
```
Comment (digitalfishpond) [inline docs/user-guide/troubleshooting.md]: ...successfully **passed authentication**, then... 
...request to **the** Dashboard **backend**. ... 
... and **will have** to  authenticate with apiserver...

**Comment by cheld** [comment]
@digitalfishpond
 
Thanks. Please merge

**Pull request merged**